utilities/openemr-cmd/openemr-cmd

```diff
@@ -24,267 +30,292 @@ get_container_names() {
+  run_devtools_in_docker "${CONTAINER_ID}" import-random-patients "$@"
 }
 
 register-oauth2-client-demo(){
-  $DOCKER_EXEC_CMD $CONTAINER_ID $SHELL_TYPE "$DEV_TOOLS register-oauth2-client-demo $1"
+  # Use the devtools function, passing all arguments ($@)
+  run_devtools_in_docker "${CONTAINER_ID}" register-oauth2-client-demo "$@"
 }
 
 register-oauth2-client(){
-  if [ $# == 1 ]
-  then
-    $DOCKER_EXEC_CMD $CONTAINER_ID $SHELL_TYPE "$DEV_TOOLS register-oauth2-client $1"
-  else
-    $DOCKER_EXEC_CMD $CONTAINER_ID $SHELL_TYPE "$DEV_TOOLS register-oauth2-client"
-  fi
+  # Use the devtools function, passing all arguments ($@)
+  run_devtools_in_docker "${CONTAINER_ID}" register-oauth2-client "$@"
 }
 
 set-swagger-to-multisite(){
-    if [ $# == 1 ]
-    then
-      $DOCKER_EXEC_CMD $CONTAINER_ID $SHELL_TYPE "$DEV_TOOLS set-swagger-to-multisite $1"
-    else
-      $DOCKER_EXEC_CMD $CONTAINER_ID $SHELL_TYPE "$DEV_TOOLS set-swagger-to-multisite"
-    fi
+  # Use the devtools function, passing all arguments ($@)
+  run_devtools_in_docker "${CONTAINER_ID}" set-swagger-to-multisite "$@"
 }
 
 set-webroot(){
-    if [ $# == 1 ]
-    then
+    local sed_command
+    if [[ $# == 1 ]]; then
       # assume the openemr parameter was given, so set openemr webroot
       echo "changing webroot to openemr"
-      $DOCKER_EXEC_CMD $CONTAINER_ID $SHELL_TYPE "sed -i 's@^DocumentRoot /var/www/localhost/htdocs.*@DocumentRoot /var/www/localhost/htdocs@g' /etc/apache2/conf.d/openemr.conf"
+      sed_command="sed -i 's@^DocumentRoot /var/www/localhost/htdocs.*@DocumentRoot /var/www/localhost/htdocs@g' /etc/apache2/conf.d/openemr.conf"
     else
       # set to blank, so set blank webroot
       echo "changing webroot to blank"
-      $DOCKER_EXEC_CMD $CONTAINER_ID $SHELL_TYPE "sed -i 's@^DocumentRoot /var/www/localhost/htdocs.*@DocumentRoot /var/www/localhost/htdocs/openemr@g' /etc/apache2/conf.d/openemr.conf"
+      sed_command="sed -i 's@^DocumentRoot /var/www/localhost/htdocs.*@DocumentRoot /var/www/localhost/htdocs/openemr@g' /etc/apache2/conf.d/openemr.conf"
     fi
+    # Use the function for arbitrary commands
+    run_shell_command_in_docker "${CONTAINER_ID}" "${sed_command}"
     # restart the docker
     echo "restarting openemr docker so the apache configuration change will take effect"
-    docker restart "$CONTAINER_ID"
+    docker restart "${CONTAINER_ID}"
 }
 
 docker-pull-image(){
+    local docker_images
     docker_images=$(docker ps --format "table {{.Image}}")
-    for IMAGE in $docker_images
-    do
-    if [ $IMAGE != "IMAGE" ] 
-    then
-      read -e -p "Do you wish to pull $IMAGE ? (y) " RESP
-      if [ "$RESP" = "y" ]; then 
-        docker pull "$IMAGE";
-      else 
-        echo "Skipping $IMAGE";
+    local IMAGE RESP
+    for IMAGE in ${docker_images}; do
+      if [[ "${IMAGE}" != "IMAGE" ]]; then
+        # Use -r with read
+        read -r -e -p "Do you wish to pull ${IMAGE} ? (y) " RESP
+        if [[ "${RESP}" == "y" ]]; then
```

Comment:
```suggestion
        if [[ "${RESP,,}" = y ]]; then
```

the ,, makes it case insensitive, and the double `==` provides no advantage over, `=`. (IMO we should avoid bashisms unless they provide some advantage. Unfortunately once shellcheck knows this is bash, it doesn't flag `==`.)